Callbook Info

aa5sh opened the pull request:
Added a context menu item to the logbook widget to to allow for selected entries to fill missing information from online callbook source.  It will not overwrite information that is there, just append any missing information.  I am mostly interested in Email address and Name but figured I would add more information that others may find usefull.  It is written to limit to 100 updates at a time.

<img width="328" alt="image" src="https://github.com/user-attachments/assets/b0dc1a44-a38d-4012-a6f2-cd7f0e493eb1" />
<img width="360" alt="image" src="https://github.com/user-attachments/assets/0cb68e50-473a-49a2-bbad-62191e924c7d" />

kyleboyle commented:
Cool.. I  would suggest changing the icon to something different - perhaps the same icon which is used for the save button.

aa5sh commented:
Yeah, I wasn't even thinking about that.  Likely can just remove it like
most of the rest of them.

On Sun, Dec 22, 2024 at 1:20 PM Kyle Boyle ***@***.***> wrote:

> Cool.. I would suggest changing the icon to something different - perhaps
> the same icon which is used for the save button.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/foldynl/QLog/pull/557#issuecomment-2558565675>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AUEEMXSKVIFZNWBOI2C3BYD2G4GHDAVCNFSM6AAAAABUBBONMWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDKNJYGU3DKNRXGU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

foldynl commented:
I kept the ideas but modified the code. There were several issues that needed to be resolved. I also added a ProgressDialog. I believe we don't want to perform a Callbook lookup while simultaneously allowing operator to modify the same QSOs. It's also important to provide users some feedback that the progress is still in-progress. Modified code in BatchLookup branch.

aa5sh commented:
Thank you.  I tried a couple times to add a progress bar but I was getting unexpected results so I thought it all away and started over.  I want to take a look at the changes so I can try to figure out what I was doing wrong.

Thank you

> On Dec 25, 2024, at 12:52 PM, Ladislav ***@***.***> wrote:
> 
> 
> I kept the ideas but modified the code. There were several issues that needed to be resolved. I also added a ProgressDialog. I believe we don't want to perform a Callbook lookup while simultaneously allowing operator to modify the same QSOs. It's also important to provide users some feedback that the progress is still in-progress. Modified code in BatchLookup branch.
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/foldynl/QLog/pull/557#issuecomment-2561975827>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AUEEMXRGMP6CN6SVIWLX6XD2HL5FVAVCNFSM6AAAAABUBBONMWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDKNRRHE3TKOBSG4>.
> You are receiving this because you authored the thread.
>

aa5sh commented:
Ok really dumb question here.  I was taking a look and it didn't seem to be working.  I added a qWarning duplicating these lines and it started to work.  Since you are starting with the qCDebug(runtime) would that make it only update in runtime and not when running from qt creator?    ![image](https://github.com/user-attachments/assets/519b07b1-4050-4fbd-9c9f-af387f8287e5)

aa5sh commented:
I am still trying to best way to debug and all and I feel I am missing something I haven't figured out just yet.  Right now I just put qWarning's everywhere to print out statements.  I need to watch/read up on debuging because there is likely a way better way I am missing just yet.

kyleboyle commented:
runtime is a logging category so-to-speak.

To debug, set breakpoints on lines of code and execute the debug target instead of run:
![Screenshot 2024-12-25 184902](https://github.com/user-attachments/assets/3953ae81-5372-4c10-af07-85906e06f40e)
when a breakpoint hits you can inspect the stack, variable values, etc using the various debug tool panels.

aa5sh commented:
Thanks wished I would have asked or looked earlier.  I figured it was something simple I just hadn’t tried.

I just been playing old school and putting those qWarning’s everywhere.  I’m used to at work the language I write in is a proprietary one and there is no real time debugger.

> On Dec 25, 2024, at 4:51 PM, Kyle Boyle ***@***.***> wrote:
> 
> 
> runtime is a logging category so-to-speak.
> 
> To debug, set breakpoints on lines of code and execute the debug target instead of run:
> Screenshot.2024-12-25.184902.png (view on web) <https://github.com/user-attachments/assets/3953ae81-5372-4c10-af07-85906e06f40e>
> when a breakpoint hits you can inspect the stack, variable values, etc using the various debug tool panels.
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/foldynl/QLog/pull/557#issuecomment-2562016486>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AUEEMXQ3Q74GONH5ZLVYAMD2HMZFJAVCNFSM6AAAAABUBBONMWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDKNRSGAYTMNBYGY>.
> You are receiving this because you authored the thread.
>

aa5sh commented:
Ok I built a test copy and used a backup db.  When I do a test it will add the name but not the email address.  But oddly enough if I add the line for the qWarning like in my screenshot above it will add the email address.

foldynl commented:
more about QCDebug on[ WIki](https://github.com/foldynl/QLog/wiki/Debug-Log-Level). You can enabled/disable selected modules based on setting of QT_LOGGING_RULES. In your case 

```
QT_LOGGING_RULES=qlog.ui.logbookwidget.*=true ./qlog
```

aa5sh commented:
Thanks.  But I guess the question I still have is for instance for this contact with NP3DM with missing information:

￼

If I right click and run the routine with starting the application without debugging like this from command line:
./qlog.app/Contents/MacOS/qlog --namespace=testing123

It will not update missing information.  But if I start it with the debuging flag:
QT_LOGGING_RULES=qlog.ui.logbookwidget.*=true ./qlog.app/Contents/MacOS/qlog --namespace=testing123

It does update it.  I guess my question is should there be a line with:
	setModeData(id, callbookValue);

On it besides in the qCDebug line?  It looks like it is only running during debug session.

Hope that makes sense.

MIchael

> On Dec 26, 2024, at 3:27 AM, Ladislav ***@***.***> wrote:
> 
> 
> more about QCDebug on WIki <https://github.com/foldynl/QLog/wiki/Debug-Log-Level>. You can enabled/disable selected modules based on setting of QT_LOGGING_RULES. In your case
> 
> QT_LOGGING_RULES=qlog.ui.logbookwidget.*=true ./qlog
> —
> Reply to this email directly, view it on GitHub <https://github.com/foldynl/QLog/pull/557#issuecomment-2562361500>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AUEEMXVJMFS26DUCK6Z64KL2HPDZLAVCNFSM6AAAAABUBBONMWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDKNRSGM3DCNJQGA>.
> You are receiving this because you authored the thread.
>

kyleboyle commented:
Yeah that's probably a bug - I'm not sure how qt handles suppression of logging statements when the rule isn't enabled... also setModeData should be renamed to setModelData.

aa5sh commented:
setModeData is a function declared higher up in the the updateQSORecordFromCallbook function.  So I think that is correct but I think it should be adjusted to be like this with the setModData on the next line.
`
void LogbookWidget::updateQSORecordFromCallbook(const QMap<QString, QString>& data)
{
    FCT_IDENTIFICATION;

    qCDebug(function_parameters) << data;

    auto getCurrIndexColumnValue = [&](const LogbookModel::ColumnID id)
    {
        return model->data(model->index(currLookupIndex.row(), id), Qt::EditRole).toString();
    };

    auto setModeData = [&](const LogbookModel::ColumnID id, const QVariant &value)
    {
        return model->setData(model->index(currLookupIndex.row(), id), value, Qt::EditRole);
    };

    if ( getCurrIndexColumnValue(LogbookModel::COLUMN_CALL) != data.value("call"))
    {
        qWarning() << "Callsigns don't match - skipping. QSO " << model->data(model->index(currLookupIndex.row(), LogbookModel::COLUMN_CALL), Qt::DisplayRole).toString()
                   << "data " << data.value("call");
        return;
    }

    const QString fnamelname = QString("%1 %2").arg(data.value("fname"),
                                                    data.value("lname"));

    const QString &nameValue = getCurrIndexColumnValue(LogbookModel::COLUMN_NAME_INTL);
    const LogbookModel::EditStrategy originEditStrategy = model->editStrategy();

    model->setEditStrategy(QSqlTableModel::OnManualSubmit);

    if ( nameValue.isEmpty()
         || data.value("name_fmt").contains(nameValue)
         || fnamelname.contains(nameValue)
         || data.value("nick").contains(nameValue) )
    {
        QString name = data.value("name_fmt");
        if ( name.isEmpty() )
            name = ( data.value("fname").isEmpty() && data.value("lname").isEmpty() ) ? data.value("nick")
                                                                                    : fnamelname;
        setModeData(LogbookModel::COLUMN_NAME_INTL, name);
    }

    auto setIfEmpty = [&](const LogbookModel::ColumnID id,
                          const QString &dataFieldID,
                          bool containsEnabled = false,
                          bool forceReplace = false)
    {
        const QString &callbookValue = data.value(dataFieldID);

        if ( callbookValue.isEmpty() )
            return;

        const QString &columnValue = getCurrIndexColumnValue(id);

        if ( columnValue.isEmpty()
             || (forceReplace && callbookValue != columnValue)
             || (containsEnabled
                  && callbookValue.contains(columnValue)
                  && callbookValue != columnValue) )
        {
            qCDebug(runtime) << "Changing"
                             << dataFieldID << callbookValue;
            setModeData(id, callbookValue);
        }
    };

    setIfEmpty(LogbookModel::COLUMN_GRID, "gridsquare", true);
    setIfEmpty(LogbookModel::COLUMN_QTH_INTL, "qth");
    setIfEmpty(LogbookModel::COLUMN_DARC_DOK, "dok");
    setIfEmpty(LogbookModel::COLUMN_IOTA, "iota");
    setIfEmpty(LogbookModel::COLUMN_EMAIL, "email");
    setIfEmpty(LogbookModel::COLUMN_COUNTY, "county");
    setIfEmpty(LogbookModel::COLUMN_QSL_VIA, "qsl_via");
    setIfEmpty(LogbookModel::COLUMN_WEB, "url");
    setIfEmpty(LogbookModel::COLUMN_STATE, "us_state");
    setIfEmpty(LogbookModel::COLUMN_ITUZ, "ituz", false, true); // always replace if different
    setIfEmpty(LogbookModel::COLUMN_CQZ, "cqz", false, true);   // always replace if different
    model->submitAll();

    model->setEditStrategy(originEditStrategy);
}
`

kyleboyle commented:
Yes, I was merely dropping a note for Lada about the fact that the 'L' is missing on the function name, it should be called setModelData because it is changing the cell values in the table's model :)

aa5sh commented:
Ah ok sorry about that.  Makes sense.

> On Dec 26, 2024, at 5:34 PM, Kyle Boyle ***@***.***> wrote:
> 
> 
> Yes, I was merely dropping a note for Lada about the fact that the 'L' is missing on the function name, it should be called setModelData because it is changing the cell values in the table's model :)
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/foldynl/QLog/pull/557#issuecomment-2563170109>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AUEEMXRC2FLM5UYIEFAE4C32HSHA7AVCNFSM6AAAAABUBBONMWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDKNRTGE3TAMJQHE>.
> You are receiving this because you authored the thread.
>

foldynl commented:
Thanks for the info. Branch updated; Typo fixed, method moved outside of qcdebug.
